node/rpcstack.go

```diff
@@ -73,9 +73,10 @@ type httpServer struct {
 	mu       sync.Mutex
 	server   *http.Server
 	listener net.Listener // non-nil when server is running
+	ready    bool
+	shutdown chan struct{} // Channel to wait for termination notifications
```

Comment:
I think we should avoid adding channels to this codebase; the channel management is already very poor. I imagine what you want here can be accomplished with ` sync.WaitGroup` instead. In `newHTTPServer`, instead of using `make(chan struct{})` you can just say `shutdownWG.Add(1)`. In `stop` you can now simply say `shutdownWG.Done()`. And, in `wait` you can just call `shutdownWG.Wait()` without all the nil-checking logic.